docs/packages/context-menu/Context-menu-manager.md

```diff
@@ -0,0 +1,345 @@
+  - [Delete a group](#Delete a group)
+  - [Add item](#Add item)
+  - [Update item](#Update item)
+  - [Delete item](#Delete item)
+  - [Get config](#Get config)
+- [Understanding the usage flow]()
+  - [Registration of new items](#Registration of new items)
+  - [Triggering the context menu flow](#Triggering the context menu flow)
+- [Example of usage]()
+
+## Introduction
+The context menu manager provides a convenient way for managing the context menu structure.
+The manager is responsible for building a tree of nested groups/items that forms the actual structure of the context menu, as presented in the UI.   
+The manager is based on CRUD principles while providing simple functions for adding, updating, deleting, and in the end - provides a function for reading the main tree that have been created.
+
+### Context menu UI behavior
+For better understanding the structure and the manager, lets see some important UI notes:
+
+* The context menu displays a list of items, by default each item provides an action when clicking on it.
+
+* Items can provide action also on hover (mouseEnter).
+
+* If an item contains children groups, hovering on it will open a sub menu, and clicking on it won't trigger action.
+
+* Groups themselves are not visually displayed, they just displays their items and separate between them.
+
+* A group can display a title, if a title is specified.
+
+* Items of a group are displayed together, separated from other group items with a line separator.
+
+* Nested groups are displayed as a sub-menu aside the main context-menu.
+
+### Context Menu config structure
+The context menu config structure is built as a tree of groups & items.
+The root is based on groups, each group contains one or more item (inside its `items[]` property).
+Items can also have children, inside their `groups[]` property. those children are *nested groups*.
+
+### Structure tree
+
+```TS
+interface Group {
+	id: string; 				// Must be unique.
+	title: string;				// If exists, displayed as a group title without action to trigger.
+	disabled: Function | boolean;		// If true, the entire group is Displayed but disabled.
+	parentItemId: string;			// Id of the parent (a valid item id). if used - creates a nested menu.
+	items: Item[]; 				// List of items.
+};
+
+interface Item {
+	id: string;				// Must be unique.
+	label: string;				// Describes the action.
+	icon: Function | string;		// Can be a callback for checkbox effect.
+	shortCut: string;			// ShortCut symbols. 
+	badge: Badge;				// Creates a visual badge for special indications.
+	disabled: Function | boolean;		// If true, displayed but disabled.
+	action: Function;			// A callback to be triggered on-click.
+	onMouseEnter: Function;			// A callback to be triggered on-mouse-enter.
+	onMouseLeave: Function;			// A callback to be triggered on-mouse-leave.
+	groups: Group[];			// List of groups (for nested menus).
+};
+
+interface Badge {
+	backgroundColor: string;		// Background color of the badge.
```

Comment:
what if I want gradient?

Reply:
We support only a simple badge.

Reply:
for now. In the future we might want a custom one

Reply:
I prefer not to allow JSX etc... the budge usage is specific and precise.